### DISCUSSION TITLE
Winlogbeat to Heavynode?

### DISCUSSION THREAD

**Discussion by ben-sec** [Security-Onion-Solutions/securityonion#9996] [Unsupported Versions] [answered]
Hello!

I have a new remote Heavynode in my distributed environment and want to collect Windows event logs there. Can I send them directly to the Heavynode or do they have to go to my central Manager node? Thanks in advance!

Cheers, Ben

**Comment by InfosecGoon** [comment]
You should be able to send them to the Heavy Node, though you might have to open up the firewall settings to allow the inbound traffic. How are you transporting the logs? Winlogbeat?

**Reply by ben-sec** [reply]
Thanks, I will give at try with winlogbeat.

**Reply by ben-sec** [reply]
Now I found [this](https://docs.securityonion.net/en/2.3/ingest.html?highlight=pipeline#heavy) about ingest pipelines in the SO documentation and there is no word about Winlogbeat when it comes to Heavy Nodes. I'm also wondering where I can find the logstash configuration file on the Heavy Node that I would have to change in order to support SSL encryption?

**Reply by InfosecGoon** [reply]
You would be sending from Winlogbeat on a Windows host to the Logstash receiver on the Heavy Node. By default, this is on port 5044 for plaintext, but if you want to do it over an encrypted tunnel that would be port 5644. The certificate information should be in the configuration file for filebeat (/opt/so/conf/filebeat/etc/filebeat.yml) on the Heavy Node.

**Reply by ben-sec** [reply]
If I run so-allow on my Heavy Node to open port 5044 for Winlogbeat clients I get the following error:
```
Traceback (most recent call last):
  File "/sbin/so-firewall", line 410, in <module>
    main()
  File "/sbin/so-firewall", line 400, in main
    code = cmd(options, args[1:])
  File "/sbin/so-firewall", line 321, in includehost
    result = addIp(args[0], args[1], 'insert')
  File "/sbin/so-firewall", line 119, in addIp
    content = loadYaml(hostgroupsFilename)
  File "/sbin/so-firewall", line 77, in loadYaml
    file = open(filename, "r")
FileNotFoundError: [Errno 2] No such file or directory: '/opt/so/saltstack/local/salt/firewall/hostgroups.local.yaml'
```
Is so-allow not working on a Heavy Node?

**Reply by InfosecGoon** [reply]
So-allow only runs on a Manager/Standalone node, because it updates the Salt configuration files there.

If you want to open up port 5044 on a minion node (like a Heavy node), you'll need to make the appropriate modifications to the node's minion pillar file. Something like this, but with beats_endpoint as the hostgroup instead of syslogtosensor1.

https://docs.securityonion.net/en/2.3/firewall.html#allow-hosts-to-send-syslog-to-a-sensor-node

**Reply by ben-sec** [reply]
I have added the IP of my heavynode to the hostgroup and added
```
firewall:
  assigned_hostgroups:
    chain:
      DOCKER-USER:
         hostgroups:
           beats_endpoint:
             portgroups:
               - portgroups.beats_5044
```
to my_minion_heavynode.sls, but there is no data comming in. Where do I find the firewall logs to verify if port still 5044 gets blocked (I start off to send them in plain text to keep things simple)?

**Reply by ben-sec** [reply]
I found the firewall logs and it seems that the first few Winlogbeat packets got dropped, but then the correct firewall configuration kicked in and I don't see any dropped packets for dst-port 5044 anymore. But I don't see any Winlogbeat data from my test client in SO. Other clients, who are sending their data to the Manager node are working fine.  Ideas?

**Reply by ben-sec** [reply, marked as answer]
The logstash ssl configuration of my manager node has been distributed to the heavynode and it was expecting ssl traffic from winlogbeat. After setting up the ssl configuration in winlogbeat.yaml everything is working fine!
